### DISCUSSION TITLE
Missing Auto instrumented metrics from Netty using grpc-java

### DISCUSSION THREAD

**Discussion by bcarter97** [open-telemetry/opentelemetry-java-instrumentation#11311] [Q&A] [answered]
## Description

Hello, I am using grpc-java with the grpc-netty implementation, and I was expecting to see HTTP Server metrics auto-instrumented from Netty as it is marked as [supported](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md) for client/server metrics. I can see that rpc metrics are being instrumented, but no http metrics. Additionally I have disabled span suppression and also can't see any Netty traces.

- grpc-netty - 1.61.0 (transistively uses Netty 4.1.100.Final)
- opentelemetry-javaagent - 2.3.0

I see some logs such as:

```text
Loading instrumentation netty [class io.opentelemetry.javaagent.instrumentation.netty.v4_1.NettyInstrumentationModule]
...
Transformed io.grpc.netty.NettyChannelBuilder -- jdk.internal.loader.ClassLoaders...
...
Applying instrumentation: netty [class io.opentelemetry.javaagent.instrumentation.netty.v4_1.NettyInstrumentationModule] on jdk.internal.loader.ClassLoaders...
...
```

The only metrics I can see are similar to `rpc_server_duration_` and `rpc_client_duration_`

**Comment by laurit** [comment, marked as answer]
probably because our netty instrumentation does not support http/2

**Reply by bcarter97** [reply]
interesting, is it planned to be supported?

I can see the following issues, but nothing directly relating to implementing http2. 

- https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8927
- https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/3227

**Reply by laurit** [reply]
> is it planned to be supported?

I definitely hope to see tests running with http/2, but I can not say anything about when this might happen.
